tests/celery_tests.py

```diff
@@ -174,8 +177,23 @@ def test_run_sync_query_cta(self, ctas_method):
         )
         # provide better error message
         self.assertEqual(QueryStatus.SUCCESS, result["query"]["state"], msg=result)
-        self.assertEqual([], result["data"])
-        self.assertEqual([], result["columns"])
+
+        expected_result = []
+        if backend == "presto":
+            expected_result = (
+                [{"rows": 1}] if ctas_method == CtasMethod.TABLE else [{"result": True}]
+            )
+        self.assertEqual(expected_result, result["data"])
+        expected_columns = []
+        if backend == "presto":
+            expected_columns = [
+                {
+                    "name": "rows" if ctas_method == CtasMethod.TABLE else "result",
+                    "type": "BIGINT" if ctas_method == CtasMethod.TABLE else "BOOLEAN",
+                    "is_date": False,
+                }
+            ]
```

Comment:
This also runs the risk of becoming convoluted if we introduce db-specific logic for all supported dbs. As this logic is repeated below, I think there is merit to centralizing this somewhere, too, but `db_engine_specs` probably isn't the right place for test assertion related stuff. I'm open to suggestions, but feel we can probably leave this as-is until we come up with a more scalable solution.

Reply:
agree, will tackle it in the upcoming PRs, do not have clear solution yet.
most likely it would be possible to introduce TestDBSpec and hide that complexity there.